src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -5,7 +5,7 @@
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T> {
```

Comment:
we dont want to work with other classes rather Machine-like childs 
```suggestion
public interface MachineService<T extends Machine> {
```